README.md

```diff
@@ -5,26 +5,82 @@
 + Explain the order in which lines of code get executed
 
 ## Callback Order of Operations
+Knowing when a line of code will run is easy if those lines are written like this:
 
-Order of operations can be confusing, code not run line by line like in ruby, second line doesn't fire until second line finishes, even if that means getting an API request, things won't happen until the data is retrieved.
+```js
+console.log(1);
+console.log(2);
+console.log(3);
+```
+
+We start from line #1 and execute each statement in order. If we run this code we'll see the following in our console:
+
+```
+1
+2
+3
+```
+
+Nothing special right? Each console statement is executed one after another. When we introduce callbacks, the order of operation changes. If we were to take our previous example and place the `console.log(2)` inside of the callback of an Ajax request:
 
 ```js
-console.log(1)
-$.getJSON("http://api.giphy.com/v1/gifs/search?q=fat+cat&api_key=dc6zaTOxFJmzC", function(response){
-        //do something with the response data here!
-console.log(2)
+console.log(1);
+$.getJSON("https://api.github.com/repos/rails/rails/commits", function(response){
+  console.log(2);
 });
-console.log(3)
+console.log(3);
 ```
 
-give more examples - some with success and some with failure and explain what code fires when
+The console output would change to this:
 
-## Instructions
-have students write code that has alerts or console.log print in order that you set - like with numbers `1` before `2` before `3`
+```
+1
+3 // The 3 is printed before the 2!!
+2
+```
 
-or you can have them make a complete sentence - `i love javascripe`: `i` before `love` before `javascript` within their ajax get request
+What happend, why are things out of order now?! The answer is simple, `console.log(2)` won't run until the `$.getJSON` completes. Our code executes in order except when we reach the Ajax request. The order now looks like this.
+
+- `console.log(1)` executes.
+- We create an Ajax request with `$.getJSON`.
+  - Our callback contains `console.log(2)` but `console.log(2)` isn't executed just yet.
+- `console.log(2)` executes.
+- Our ajax request completes and `console.log(2)` executes.
+
+## When something goes wrong
+
+As we learned in previous lessions, not all Ajax requests complete successfully. Sometimes things go wrong and errors happen. When we add both success and error callbacks, order of operations is a little more confusing. In this next example, we run `console.log("Success")` only if the Ajax request completes successfully but if it fails, we only run `console.log("Fail")`.
+
+```js
+console.log(1);
+$.getJSON("https://api.github.com/repos/rails/rails/commits", function(response){
+  console.log("Success");
+}).fail(function(response){
```

Comment:
maybe briefly explain how need to explicitly set up fail callback but success runs automatically based on a sucesful response